geojson_modelica_translator/modelica/modelica_project.py

```diff
@@ -120,9 +131,45 @@ def pretty_print_tree(self) -> None:
             indent = key.count(os.path.sep)
             print(" " * indent + f"{os.path.sep} {key.replace(os.path.sep, f' {os.path.sep} ')}")
 
+    def get_model(self, model_name: Union[Path, str]) -> Model:
+        """Return the model object based on the based string name. The model
+        name should be in the format that Modelica prefers which is period(.)
+        delimited.
+
+        Args:
+            model_name (str): Name of the model to return, in the form of . delimited
+
+        Raises:
+            Exception: Various exceptions if the model is not found or the file type is incorrect
+
+        Returns:
+            Model: The Modelica Builder model object
+        """
+        # check if the last 3 characters are .mo. The path should originally be
+        # a period delimited path.
+        model_name = str(model_name)
+        if model_name[-3:] == '.mo':
+            raise Exception(f"Model name should not have the .mo extension: {model_name} ")
+
+        # convert the model_name to the path format
+        model_name = Path(model_name.replace('.', os.path.sep))
+
+        # now add on the extension
+        model_name = model_name.with_suffix('.mo')
+
+        if self.file_data.get(str(model_name)) is None:
+            raise Exception(f"ModelicaPackage does not contain a {model_name} model")
+        else:
+            # verify that the type of file is model
+            model = self.file_data[str(model_name)]
+            if model.file_type != ModelicaFileObject.FILE_TYPE_MODEL:
+                raise Exception(f"Model is a package file, not a model: {model_name}")
+
+            return self.file_data[str(model_name)].object
+
     def save_as(self, new_package_name: str, output_dir: Path = None) -> None:
         """method to save the ModelicaProject to a new location which
-        requires a new path name and updating all of the "within" statements
+        requires a new path name and updating all of the within statement
```

Comment:
```suggestion
        requires a new path name and updating all of the "within" statements
```

To not overwrite the wee little change I made in the previous PR